sno/working_copy/gpkg.py

```diff
@@ -178,9 +168,48 @@ def delete(self):
             f.unlink()
 
     def is_created(self):
+        """
+        Returns true if the GPKG file referred to by this working copy exists.
+        Note that it might not be initialised as a working copy - see self.is_initialised.
+        """
         return self.full_path.is_file()
 
-    def create(self):
+    def is_initialised(self):
+        """
+        Returns true if the GPKG working copy is initialised -
+        the GPKG file exists and has the necessary gpkg_sno tables - state and tracking.
+        """
+        if not self.is_created():
+            return False
+        with self.session() as db:
+            dbcur = db.cursor()
+            dbcur.execute(
+                f"""
+                SELECT count(*) FROM sqlite_master
+                WHERE type='table' AND name IN ({self.STATE_TABLE}, {self.TRACKING_TABLE});
```

Comment:
`self.STATE_TABLE` and `self.TRACKING_TABLE` are identifier-quoted strings, does this work? It seems like it probably shouldn't

Reply:
It works because the engine we're using doesn't have identifier specific quoting anyway - gpkg.ident() just adds double quotes and escapes any existing double quotes, which is fine for the name of a column, or for a string. Tests would fail if it didn't work.

Will leave this as-is for now - can revisit when we get to sqlalchemy.

Reply:
double-quotes _is_ identifier-specific quoting, at least in postgres. I'm surprised it works in sqlite

Reply:
https://sqlite.org/lang_keywords.html

> If you want to use a keyword as a name, you need to quote it. There are four ways of quoting keywords in SQLite:
>
> * `'keyword'` A keyword in single quotes is a string literal.
> * `"keyword"` A keyword in double-quotes is an identifier.
> * `[keyword]` A keyword enclosed in square brackets is an identifier. This is not standard SQL. This quoting mechanism is used by MS Access and SQL Server and is included in SQLite for compatibility.
> * `` `keyword` `` A keyword enclosed in grave accents (ASCII code 96) is an identifier. This is not standard SQL. This quoting mechanism is used by MySQL and is included in SQLite for compatibility.
> 
> For resilience when confronted with historical SQL statements, SQLite will sometimes bend the quoting rules above:
> * If a keyword in single quotes (ex: `'key'` or `'glob'`) is used in a context where an identifier is allowed but where a string literal is not allowed, then the token is understood to be an identifier instead of a string literal.
> * If a keyword in double quotes (ex: `"key"` or `"glob"`) is used in a context where it cannot be resolved to an identifier but where a string literal is allowed, then the token is understood to be a string literal instead of an identifier.

🤷